### PR TITLE
Fix validation errors about using unsupported linear filtering

### LINKED ISSUE #923: Framework and most samples are assuming linear filtering is supported for all formats, but should be checking
Most of the samples, and parts of the framework, just build samplers using linear filtering and mipmap-modes. This is only guaranteed to be supported for a handful of formats. On my hardware this results in a large number of validation errors.

I have a (fairly large) PR ready that addresses these issues, which I'll push shortly.

### PR DESCRIPTION
**Pull request by gary-sweet**
## Description

Most of the samples, and parts of the framework, just build samplers using linear filtering and mipmap-modes. This is only guaranteed to be supported for a handful of formats. On my hardware this results in a large number of validation errors.

Fixes #923

## General Checklist:

Please ensure the following points are checked:

- [x] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [x] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [x] I have commented any added functions (in line with Doxygen)
- [x] I have commented any code that could be hard to understand
- [x] My changes do not add any new compiler warnings
- [x] My changes do not add any new validation layer errors or warnings
- [x] I have used existing framework/helper functions where possible
- [x] My changes do not add any regressions
- [x] I have tested every sample to ensure everything runs correctly (those that run for me at least)
- [x] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [x] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [x] My changes build on Linux

### CODE REVIEW

File: framework/core/hpp_physical_device.cpp [hunk 1]
```diff
@@ -138,5 +138,12 @@ vk::PhysicalDeviceFeatures &HPPPhysicalDevice::get_mutable_requested_features()
 	return requested_features;
 }
 
+vk::FormatProperties HPPPhysicalDevice::get_format_properties(vk::Format format) const
+{
+	vk::FormatProperties format_properties;
+	handle.getFormatProperties(format, &format_properties);
+	return format_properties;
```
Comment (asuessenbach) [inline framework/core/hpp_physical_device.cpp]: Just call
`	return handle.getFormatProperties(format);
`
Or maybe even skip this get_format_properties-function and just use
`gpu.get_handle().getFormatProperties(format)`, instead

File: framework/gltf_loader.cpp [hunk 2]
```diff
@@ -647,13 +648,19 @@ sg::Scene GLTFLoader::load_scene(int scene_index)
...
+
+			texture->set_sampler(has_linear_filter ? *default_sampler_linear : *default_sampler_nearest);
 		}
 
 		scene.add_component(std::move(texture));
 	}
 
-	scene.add_component(std::move(default_sampler));
+	scene.add_component(std::move(default_sampler_linear));
+	scene.add_component(std::move(default_sampler_nearest));
```
Comment (asuessenbach) [inline framework/gltf_loader.cpp]: Maybe only add the sampler(s), that are actually used.

File: samples/api/hpp_texture_loading/hpp_texture_loading.cpp [hunk 3]
```diff
@@ -488,35 +488,19 @@ void HPPTextureLoading::load_texture()
...
-	{
-		// The device does not support anisotropic filtering
-		sampler.maxAnisotropy    = 1.0;
-		sampler.anisotropyEnable = false;
-	}
-	sampler.borderColor = vk::BorderColor::eFloatOpaqueWhite;
-	texture.sampler     = device.createSampler(sampler);
+
+	texture.sampler = vkb::common::create_sampler(get_device()->get_gpu().get_handle(), get_device()->get_handle(),
+	                                              format, vk::Filter::eNearest, vk::SamplerAddressMode::eClampToEdge,
```
Comment (asuessenbach) [inline samples/api/hpp_texture_loading/hpp_texture_loading.cpp]: You unconditionally downgraded the sampler here from linear to nearest!

**Review by asuessenbach — COMMENTED** [review]
Just a few minor issues.

File: framework/core/hpp_physical_device.cpp (hunk 1 above)
Comment (gary-sweet) [reply framework/core/hpp_physical_device.cpp]: I'll remove it, and use your second suggestion

File: framework/gltf_loader.cpp (hunk 2 above)
Comment (gary-sweet) [reply framework/gltf_loader.cpp]: Done

File: samples/api/hpp_texture_loading/hpp_texture_loading.cpp (hunk 3 above)
Comment (gary-sweet) [reply samples/api/hpp_texture_loading/hpp_texture_loading.cpp]: D'oh! Copy-and-paste for the win.

**Review by SaschaWillems — DISMISSED** [review]
LGTM!

**Comment by gary-sweet** [comment]
Sorry, needed to rebase to resolve merge conflicts.

**Review by SaschaWillems — DISMISSED** [review]

File: framework/core/hpp_physical_device.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (c) 2022-2023, NVIDIA CORPORATION. All rights reserved.
+/* Copyright (c) 2022-2024, NVIDIA CORPORATION. All rights reserved.
```
Comment (asuessenbach) [inline framework/core/hpp_physical_device.cpp]: Only the copyright has changed?

File: framework/core/hpp_physical_device.h [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (c) 2023, NVIDIA CORPORATION. All rights reserved.
+/* Copyright (c) 2023-2024, NVIDIA CORPORATION. All rights reserved.
```
Comment (asuessenbach) [inline framework/core/hpp_physical_device.h]: Only the copyright has changed?

File: framework/core/hpp_physical_device.cpp (hunk 4 above)
Comment (gary-sweet) [reply framework/core/hpp_physical_device.cpp]: Fixed

File: framework/core/hpp_physical_device.h (hunk 5 above)
Comment (gary-sweet) [reply framework/core/hpp_physical_device.h]: Fixed

**Review by asuessenbach — DISMISSED** [review]

**Review by SaschaWillems — DISMISSED** [review]

**Comment by gary-sweet** [comment]
Sorry, I needed to rebase this due to merge conflicts in mobile_nerf.cpp

**Review by asuessenbach — DISMISSED** [review]

**Comment by gary-sweet** [comment]
Rebased again after yesterday's merges caused more conflicts

**Review by asuessenbach — DISMISSED** [review]

**Review by SaschaWillems — DISMISSED** [review]

**Comment by gary-sweet** [comment]
More merge conflicts I'll need to look at again - sigh.

I can't seem to get this past 2 approvals before someone else merges something.

**Comment by SaschaWillems** [comment]
That's annoying.

@marty-johnson59 : I don't think we need three approvals on this. It's just a bugfix and the code is perfectly fine. Can we merge once Gary has fixed the current merge conflicts?

**Comment by gary-sweet** [comment]
> @marty-johnson59 : I don't think we need three approvals on this. It's just a bugfix and the code is perfectly fine. Can we merge once Gary has fixed the current merge conflicts?

Thanks Sascha, that would be great if possible.

**Review by JoseEmilio-ARM — APPROVED** [review]

**Comment by marty-johnson59** [comment]
OK, I'll merge. LMK if this causes problems.

**Pull request merged**